backend/app/categories/controller.py

```diff
@@ -46,8 +46,27 @@ async def get_category_item_route(category_id: CategoryEnum, item_id: ObjectID,
 async def add_item_route(category_id: CategoryEnum, item: Union[Movie, Anime, Show, Music],
                          db: AsyncIOMotorClient = Depends(get_database)) -> ItemInResponse:
     """Add item to a given category"""
-    # TODO add item to db
-    return {"success": True}
+    if category_id not in CategoryEnum.__members__:
+        raise HTTPException(status_code=404, detail='Invalid category_id')
+
+    # creating dict for checking response body
+    classof = {
+        CategoryEnum.movies: Movie,
+        CategoryEnum.shows: Show,
+        CategoryEnum.anime: Anime,
+        CategoryEnum.books: Book,
+        CategoryEnum.music: Music
+    }
+
+    # checking if request body is of correct type
+    if not isinstance(item, classof[category_id]):
```

Comment:
Good work!
Additionally, you can change,
```python
if not isinstance(item, item_class_mapping.get(category_id, None)):
    return ItemInResponse(success=False, data=[f"Item to be added does not match {category_id} schema"])
```

Reply:
Its resulting into this error as we dont have all fields req. 
![image](https://user-images.githubusercontent.com/56212978/83974590-25606f00-a90c-11ea-97d2-74aeb642ec82.png)

Reply:
Sorry my bad, I meant to say `error=["same message"]`

Reply:
That to gave similar error as data is none there.

Reply:
And what do you mean by item_class_mapping.get(category_id,None) ?

Reply:
Instead of naming the variable `classof` name it something more appropriate like `item_class_mapping`. And `.get` is a function for dict to fetch a key. Anyways, that's not too important.